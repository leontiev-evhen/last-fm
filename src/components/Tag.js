import React from 'react';
import PropTypes from 'prop-types';

export function Tag({ name }) {
	return <div className="tag-section">{name}</div>;
}

Tag.propTypes = {
	name: PropTypes.string.isRequired,
};
