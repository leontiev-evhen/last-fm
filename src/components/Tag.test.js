import React from 'react';
import { mount } from 'enzyme';
import { Tag } from './Tag';

describe('Tag component', () => {
	const props = {
		name: 'name',
	};
	const component = mount(<Tag {...props} />);
	it('render correctly Tag component', () => {
		expect(component).toMatchSnapshot();
	});
	it('check prop name with html class', () => {
		expect(component.find('.tag-section').text()).toEqual('name');
	});
	it('check the type of name', () => {
		expect(typeof component.prop('name')).toBe('string');
	});
});
