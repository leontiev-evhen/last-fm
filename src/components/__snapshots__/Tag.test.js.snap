// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag component render correctly Tag component 1`] = `
ReactWrapper {
  Symbol(enzyme.__unrendered__): <Tag
    name="name"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "name": "name",
    },
    "ref": null,
    "rendered": Object {
      "instance": <div
        class="tag-section"
      >
        name
      </div>,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "name",
        "className": "tag-section",
      },
      "ref": null,
      "rendered": Array [
        "name",
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "name": "name",
      },
      "ref": null,
      "rendered": Object {
        "instance": <div
          class="tag-section"
        >
          name
        </div>,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "name",
          "className": "tag-section",
        },
        "ref": null,
        "rendered": Array [
          "name",
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
